name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FinTechGroupBankPDFExtractor.java

```diff
@@ -998,6 +1101,31 @@ private void addDividendeWithNegativeAmountTransaction()
                                         })
                                 ,
                                 // @formatter:off
+                                // Extag           :    13.07.2023        Bruttothesaurierung:       -32,86 USD
+                                // Valuta          :    14.07.2023       *Einbeh. Steuer     :         0,15 EUR
+                                // Zuflusstag      :    14.07.2023        Devisenkurs        :         1,118200
+                                //                                        Endbetrag          :        -0,15 EUR
+                                // @formatter:on
+                                section -> section
+                                        .attributes("fxCurrency", "gross", "currency", "exchangeRate")
+                                        .match("^.* (Bruttodividende|Bruttoaussch.ttung|Bruttothesaurierung)([:\\s]+)? \\-[\\.,\\d]+ (?<fxCurrency>[\\w]{3})$")
+                                        .match("^.* [\\*]+Einbeh\\. Steuer([:\\s]+)? (?<gross>[\\.,\\d]+) (?<currency>[\\w]{3})$")
+                                        .match("^(.* )?Devisenkurs([:\\s]+)? (?<exchangeRate>[\\.,\\d]+).*$")
+                                        .match("^.* Endbetrag([:\\s]+)? \\-[\\.,\\d]+ [\\w]{3}$")
+                                        .assign((t, v) -> {
+                                            v.put("baseCurrency", asCurrencyCode(v.get("currency")));
+                                            v.put("termCurrency", asCurrencyCode(v.get("fxCurrency")));
+
```

Comment:
Nothing wrong with this code, but why don't you match the currency directly as ```baseCurrency``` and ```termCurrency```? Then you do not have to "re-map" them in the code.

Reply:
I allowed myself to change it to:
```
ExtrExchangeRate rate = asExchangeRate(v);
type.getCurrentContext().putType(rate);

Money gross = Money.of(rate.getBaseCurrency(), asAmount(v.get("gross")));
Money fxGross = rate.convert(rate.getTermCurrency(), gross);
```